log.go

```diff
@@ -54,25 +56,63 @@ func levelToName(level Level) string {
 	panic("Unknown log level")
 }
 
+func levelToRollbarName(level Level) string {
+	switch level {
+	case Trace:
+		return "debug" // Rollbar does not have a Trace level, so setting to the lowest Rollbar level
+	case Debug:
+		return "debug"
+	case Info:
+		return "info"
+	case Warn:
+		return "warning"
+	case Error:
+		return "error"
+	}
+	panic("Unknown log level")
+}
+
 // A logger object. Use New() to construct one.
 type Logger struct {
 	Level      Level // Log messages with a level lower than this are discarded. Default level is Info
 	lj         lumberjack.Logger
 	shownError bool
+	useRollbar bool // Log messages to Rollbar
 }
 
 // Create a new logger. Filename may also be one of the special names log.Stdout and log.Stderr
 func New(filename string) *Logger {
 	l := &Logger{
-		Level: Info,
+		Level:      Info,
+		useRollbar: false,
 	}
 	l.lj.Filename = filename
 	l.lj.MaxSize = 30
 	l.lj.MaxBackups = 3
 	return l
 }
 
+// SetupRollbar sets up the Rollbar environment.
+// Rollbar is an improvement on the log scraper which sends messages to a online rollbar server.
+// The server then informs the relavent parties about the message via email immediately
+func (l *Logger) SetupRollbar(token string, repository string, version string, environment string) {
+	rollbar.SetToken(token)
+	rollbar.SetEnvironment(environment)
+	rollbar.SetServerRoot(repository)
+	rollbar.SetCodeVersion(version)
+	l.useRollbar = true
+}
+
+func (l *Logger) SendRollbarMessage(level Level, message string) {
+	if l.useRollbar {
+		rollbar.Message(levelToRollbarName(level), message)
+	}
+}
+
 func (l *Logger) Close() error {
+	if l.useRollbar {
+		rollbar.Wait() // Wait for all of the rollbar message to clear before Closing
```

Comment:
It would be annoying/SLA-impacting if a service took 2 minutes to shutdown because it was waiting for an internet connection. Not sure we should bother with this.